File: backend/src/application/repositories/ride.repository.ts
import { CreateRideData, Ride, RideStatus } from '../../domain/entities/ride.entity'

export interface RideFilters {
  userId?: string
  status?: RideStatus
}

export interface RideRepository {
  create(data: CreateRideData): Promise<Ride>
  findById(id: string): Promise<Ride | null>
  findAll(filters?: RideFilters): Promise<Ride[]>
  updateStatus(id: string, status: RideStatus, driverId?: string): Promise<Ride>
}

File: mobile-driver/lib/domain/repositories/ride_repository.dart
import '../entities/ride.dart';

abstract class RideRepository {
  Future<List<Ride>> listPending();
  Future<List<Ride>> listByDriver(String driverId);
  Future<Ride> getById(String id);
  Future<Ride> updateStatus(String id, RideStatus status, {String? driverId});
}
